src/libraries/System.Net.Security/src/System/Net/Security/Pal.Android/SafeDeleteSslContext.cs

```diff
@@ -59,13 +66,20 @@ public SafeDeleteSslContext(SslAuthenticationOptions authOptions)
 
         protected override void Dispose(bool disposing)
         {
-            if (disposing)
+            if (disposing && !_disposed)
             {
-                if (_sslContext is SafeSslHandle sslContext)
+                _disposed = true;
+
+                lock (_lock)
                 {
                     _inputBuffer.Dispose();
                     _outputBuffer.Dispose();
-                    sslContext.Dispose();
+                    _sslContext.Dispose();
+                }
+
+                if (_gcHandle.IsAllocated)
+                {
+                    _gcHandle.Free();
```

Comment:
Technically this could be called twice on the same handle which according to the docs will throw.
Maybe we should instead lock on the _disposed check and only perform the clean if we successfully write true to it on the given thread. Or alternatively interlocked exchange, but it's probably not worth the trouble.